### ISSUE TITLE
Analyze real mod version on upgrading

### ISSUE THREAD

**Issue by Dymeth** [gorilla-devs/ferium#396]
### The Problem

At this moment some plugin not upgrading using `ferium upgrade` because target version isn't specified on Curseforge mod page, but in `fabric.mod.json` target version is supported. So there is some chance that the mod will work correctly on the target version. For example:
![image](https://github.com/gorilla-devs/ferium/assets/21080646/2f97f118-c5fd-44a7-9087-98f2929c3958)
![image](https://github.com/gorilla-devs/ferium/assets/21080646/e05d3e15-72dc-484c-ba75-241540aaa96b)
This mod is works perfectly on 1.20.4.

### Your Solution(s)

I suggest trying to download the latest version of the mod regardless of the version information on CurseForge, then analyze the contents of the `fabric.mod.json` file.
If the latest mod build has already been downloaded, there is no need to download it again.
I thing this feature is applicable to all mod types: Fabric, Quilt and Forge (but with different mod-metadata files).

I realize that this approach can lead to various technical problems, so I suggest to introduce this functionality as a separate application option like `--allow-unsafe-mod-versions`

**Comment by theRookieCoder** [comment]
The problem with this is the amount of time and resources it would use up.
Remember, we have to check _a lot_ of files to determine the latest compatible version. Having to download every file (which can be up to 50mb in my experience) and unzip and parse the manifest for every possible mod loader is far too much work. Not to mention, having to download so much data for as many as 300 mods (maybe even more) is just discourteous to mod distribution sites.
In short, there's a reason mod distribution sites tag compatibilities using metadata.
